charts/schema-registry/templates/deployment.yaml

```diff
@@ -46,9 +46,9 @@ spec:
                 - sh
                 - -c
                 - netstat -ltn 2> /dev/null | grep {{ .Values.service.port }} || ss -ltn 2> /dev/null | grep {{ .Values.service.port }}
-          {{- if or .Values.authentication.enabled .Values.tls.enabled (and .Values.container .Values.container.volume) }}
+          {{- if or (and .Values.authentication.enabled (not .Values.authentication.disableBasicAuthentication)) .Values.tls.enabled (and .Values.container .Values.container.volume) }}
```

Comment:
I just had a small suggestion to maybe help on the readability of these conditionals:   

replace disableBasicAuthentication by basicAuthEnabled.   Reversing the meaning may avoid some not not logic?

Reply:
Also, using template helpers can increase readability by having some conditions condensed.  For example, the "auth enabled" and not DisableBasicAuth condition could be encapsulated in a template helper called "passwordFileNeeded"

Reply:
I think "disabling" is perhaps better because all methods are enabled by default. 
Do note that an implementation for passwordauth handler (basic) is provided with the service by default. so anyone who wishes to use the basic authentication with password file should simply enable auth and provide a password file. 

We added the flag in question on a valid suggestion from you that allows admins to make a conscious choice of disabling password authentication handler. this avoids an otherwise accidental mistake where a password file could allow unintended users access to the system.

Reply:
Got it.